### PR TITLE
Move to `loadUnaligned` and hopefully streamline some helpers

### PR DESCRIPTION
**Pull request by thomasvl**
Follow up from #1604, but realized we could also stream line some of the calling code.  Each of those follow ups is done as a commit so we could drop them if desired.

Looking at things, I think we could even merge the "FourByte"/"EightByte" into a single generic function since `MemoryLayout` can provide the size.

### CODE REVIEW

File: Sources/SwiftProtobuf/BinaryDecoder.swift [hunk 1]
```diff
@@ -212,7 +212,7 @@ internal struct BinaryDecoder: Decoder {
         guard fieldWireFormat == WireFormat.fixed32 else {
             return
         }
-        try decodeFourByteNumber(value: &value)
```
Comment (thomasvl) [inline Sources/SwiftProtobuf/BinaryDecoder.swift:215]: Guess this was a bug if we ever were built for big endian.

**Review by allevato — APPROVED** [review]
This is so much cleaner.

**Comment by thomasvl** [comment]
Added one more commit to merge into a single function.

**Review by allevato — APPROVED** [review]

**Comment by thomasvl** [comment]
@tbkka gentle ping since you were the original author of a bunch of this.

**Pull request merged**